flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartition.java

```diff
@@ -120,6 +123,7 @@ public ResultPartition(
 		this.partitionType = checkNotNull(partitionType);
 		this.subpartitions = checkNotNull(subpartitions);
 		this.numTargetKeyGroups = numTargetKeyGroups;
+		this.isManagedExternally = isManagedExternally;
```

Comment:
As mentioned above, `this.isManagedExternally = (partitionType == ResultPartitionType.BLOCKING)`

Reply:
By uncoupling this field from the field in the descriptor we are running into the real risk of the logic being out-of-sync, particularly so since the descriptor is a shared API, while the ResultPartition is an internal class.

Reply:
I mean we do not add the `isManagedExternally ` property in `ResultPartitionDeploymentDescriptor` because it is only `ResultPartitionType` relevant and we could get this property directly in `ResultPartition`.  Only if `isManagedExternally` might be generated from other properties except `ResultPartitionType` in future?